htmlbook-xsl/xrefgen.xsl

```diff
@@ -224,6 +224,28 @@
     </xsl:choose>
   </xsl:template>
 
+    <!-- Special xref-to handling for refsect1/refsect2 -->
+  <xsl:template match="h:div[@class='refsect1'] | h:div[@class='refsect2']" mode="xref-to">
+    <xsl:choose>
+      <xsl:when test="h:h6[1]">
+  <!-- Choose the first descendant h6 element with class of refsect1, if one exists, drop in text-->
+    <xsl:text>“</xsl:text><xsl:value-of select="h:h6[1]"/><xsl:text>”</xsl:text>
+      </xsl:when>
+      <xsl:otherwise>
+  <!-- Otherwise, throw warning, and print out ??? -->
+  <xsl:call-template name="log-message">
+    <xsl:with-param name="type" select="'WARNING'"/>
+    <xsl:with-param name="message">
```

Comment:
Could you update this log message to say that there's a missing refsect1 or refsect2 heading, rather than missing refname?